### ISSUE TITLE
DrawPath feature not working on ReactNative

### ISSUE THREAD

**Issue by akaipham** [galkahana/PDF-Writer#102]
```js
AbstractContentContext::TextOptions textOptions(pdfWriter.GetFontForFile(
																			RelativeURLToLocalPath(
                                                                            inTestConfiguration.mSampleFileBase,
                                                                            "TestMaterials/fonts/arial.ttf")),
																			14,
																			AbstractContentContext::eGray,
																			0);
		AbstractContentContext::GraphicOptions pathFillOptions(AbstractContentContext::eFill,
																AbstractContentContext::eCMYK,
																0xFF000000);
		AbstractContentContext::GraphicOptions pathStrokeOptions(AbstractContentContext::eStroke,
																AbstractContentContext::eRGB,
																AbstractContentContext::ColorValueForName("DarkMagenta"),
																4);

		DoubleAndDoublePairList pathPoints;

		// draw path
		pathPoints.push_back(DoubleAndDoublePair(75,640));
		pathPoints.push_back(DoubleAndDoublePair(149,800));
		pathPoints.push_back(DoubleAndDoublePair(225,640));
		cxt->DrawPath(pathPoints,pathFillOptions);
		pathPoints.clear();
```

I've followed above snippet as `https://github.com/galkahana/PDF-Writer/blob/master/PDFWriterTestPlayground/HighLevelContentContext.cpp#L64`. But it didn't display any path. I tried with `DrawCircle`, `DrawSquare` and all worked well except `DrawPath`.

**Comment by galkahana** [comment]
Not sure any of this is supposed to work on react native. Hummus is a nodejs module. React native is not. In any case, its not supported.

**Comment by akaipham** [comment]
Thanks, I got it. Because the rest API like DrawCircle, DrawSquare, .. worked so I assumed that `DrawPath` would work as well. 
Is it possible to support those API for RN ?

**Comment by TheGS** [comment]
Aside... I had to look up the implementation of **AbstractContentContext::DrawPath()** to see what it does. Calling it **DrawPolygon()** would probably make its intent more easily understood at the call site

**Comment by akaipham** [comment]
Hmm, I can't see any methods name `DrawPolygon()`. Could you show me that way ?

**Comment by akaipham** [comment]
I see the declared points are not `polygon` at all, so what do you mean by `drawPolygon`?

**Comment by galkahana** [comment]
Ok sorry . I see this is opened on pdfwriter. How are you running native code on RN, btw? 
This makrs my previous note irrelevant

**Comment by galkahana** [comment]
the path described in the code is really small. it's like 2 dots
100,200 -> 100,201 -> 100,202.

this is why you see nothing.
how about trying the example path? or tell us what you are looking to do.

[b.t.w again would love notes on how you make this work on RN...i'm sure my readers would like to know. for instance, how do you create a file, how do you make the call available via the native code. notes for android, notes for ios. if you got the time it would be of great value]

**Comment by TheGS** [comment]
You're right that the **AbstractContentContext::GraphicsOptions::close** is **false** by default. Calling **AbstractContentContext::DrawPath()** without setting it to **true** will draw a polyline instead of a polygon. Sorry for the confusion.

**Comment by akaipham** [comment]
My aim to draw a polyline so I didn't set `AbstractContentContext::GraphicsOptions::close` to true like what you state above.

**Comment by akaipham** [comment]
@galkahana This is work so far to call those methods via the native code:
```js
DoubleAndDoublePairList pathPoints   = getDoubleAndDoublePairList(lineAction, @"points");
    unsigned hexColor = hexIntFromString(lineAction[@"color"]);

    AbstractContentContext::GraphicOptions pathStrokeOptions(
        AbstractContentContext::eStroke,
		AbstractContentContext::eRGB,
		AbstractContentContext::ColorValueForName("DarkMagenta"),
		10);
    context->DrawPath(pathPoints,
                        options);
```
in file `/react-native-pdf-lib/ios/PDFPageFactory.mm`

**Comment by akaipham** [comment]
I tried to set `AbstractContentContext::GraphicsOptions::close` to true and it drawn a polygon, but when I set it back to `false`, seems like it couldn't draw a line as I expected. Any suggestions?

**Comment by TheGS** [comment]
What happens when you try to draw your polyline with **RG()** [_**setrgbcolor** for stroking operations_], **m()** [_**moveto**_], **l()** [_**lineto**_], and **S()** [_**stroke**_] directly?
What happens when you try to draw the same points as a polygon with **RG()** [_**setrgbcolor** for stroking operations_], **m()** [_**moveto**_], **l()** [_**lineto**_], and **s()** [_**closepath** and **stroke**_] directly?
That should be the difference when you change the value of **AbstractContentContext::GraphicsOptions::close**, so maybe there is a problem in **AbstractContentContext::FinishPath()** or something that it calls?

**Comment by galkahana** [comment]
RE RN on native, aha got it [for IOS]. thanks. you'll need something for Android, i assume, in addition. would appreciate notes on that too once you get there.

**Comment by akaipham** [comment]
@TheGS (sorry I can't tag you), I tried `drawPath` with `AbstractContentContext::GraphicsOptions::close` set to `false` (by default) and it drawn a line as what I expected. 
@galkahana of course, your project really interesting. Thanks !
